### ISSUE TITLE
Need a way to scale an ol.style.Icon to a destination size rather than a percentage of the original

### ISSUE THREAD

**Issue by justdave** [openlayers/openlayers#13508]
In `ol.style.Icon` we should have a way to describe the destination size of the image to scale it to, without knowing the original size of the image (because it may be coming from multiple sources which are different sizes, and we want them to all be the same size on the map).

The existing `size` property crops the icon rather than scaling it, and that appears to be the intended behavior.  The `scale` property requires you to know the original size of the image in order to scale it to the correct destination size, because it scales to a percentage of the original size (whatever that size was).

This probably needs a new property such as `destSize` or somesuch to describe the target size of the scaling instead of the percentage to scale.

See also #2720 for previous (lengthy) discussion on this topic which was marked as stale and closed because nobody ever picked it up to fix it.

**Comment by mike-000** [comment]
There is already something similar to that internal to the KML format where an icon scale of 1 is based on a width of 32 pixels regardless of the native size of the image.  It works by waiting for the image to load (if not already cached) and then rescaling, similar to 

https://codesandbox.io/s/icon-forked-o1bfll?file=/main.js

**Comment by justdave** [comment]
Does this comment count as activity?  This feature is still needed.

**Comment by mike-000** [comment]
You can do it as in the CodeSandbox above.  For something more convenient for frequent use you could create a custom class, e.g. https://codesandbox.io/s/icon-forked-uof9by?file=/main.js

**Comment by KaiVolland** [comment]
I really think this would be a nice enhancement to the IconSymbolizer. In a lot of projects i worked on i came to this point and didn't found a proper solution.

During development and use of the GeoStyler we came across this problem again:
https://github.com/geostyler/geostyler-openlayers-parser/pull/549
https://github.com/geostyler/geostyler-openlayers-parser/issues/624

I could imagine `width` and `height` (px) properties on the IconSymbolizers that behave similar to CSS (keep aspect-ratio when only one of them is given, etc.).

What do you think about this @ahocevar @tschaub @mike-000 @MoonE ?

As the Sandbox example above shows this shouldn't be to hard to implement.

**Comment by ahocevar** [comment]
Sounds good, @KaiVolland. Would you be able to provide a pull request?

**Comment by KaiVolland** [comment]
Yes i can give it a try. Not sure when i will be able to find the time for this but i'll do my best.
